sygnal/utils.py

```diff
@@ -37,17 +41,19 @@ async def twisted_sleep(delay, twisted_reactor):
 
 
 class NotificationLoggerAdapter(LoggerAdapter):
-    def process(self, msg, kwargs):
+    def process(
+        self, msg: str, kwargs: MutableMapping[str, Any]
+    ) -> Tuple[str, MutableMapping[str, Any]]:
         return f"[{self.extra['request_id']}] {msg}", kwargs
 
 
-def glob_to_regex(glob):
+def glob_to_regex(glob: str) -> Union[Pattern, Pattern[str]]:
```

Comment:
I don't understand the return type here---why can't we just say it returns a `Pattern[str]` in all cases?

Reply:
I nicked this from somewhere and now I can't remember where, but I think you are correct we can use the narrower case. I will change this.